WalletWasabi.Backend/Controllers/ChaumianCoinJoinController.cs

```diff
@@ -228,9 +228,14 @@ public async Task<IActionResult> PostInputsAsync([FromBody, Required]InputsReque
 								return BadRequest("Provided input is neither confirmed, nor is from an unconfirmed coinjoin.");
 							}
 
-							// Check if mempool would accept a fake transaction created with the registered inputs.
+							// Check if mempool would accept a fake transaction created with all of the registered inputs plus the new input.
 							// This will catch ascendant/descendant count and size limits for example.
-							var result = await RpcClient.TestMempoolAcceptAsync(new[] { new Coin(inputProof.Input, getTxOutResponse.TxOut) });
+
+							var currentAlices = round.GetAlicesBy(AliceState.InputsRegistered).ToArray();
+							var coinsToTest = currentAlices.SelectMany(alice => alice.Inputs).ToList();
+							coinsToTest.Add(new Coin(inputProof.Input, getTxOutResponse.TxOut));
+
+							var result = await RpcClient.TestMempoolAcceptAsync(coinsToTest, fakeOutputsCount: currentAlices.Length * 2);
```

Comment:
IMO too few. This is a lower bound-ish estimation. Maybe do 3x for an upper bound-ish estimation instead?

Reply:
Also it's slightly better to use `coinsToTest.Count()` instead of `currentAlices.Length`.

Reply:
I used it to avoid double enumeration.

Reply:
Randomly checked 5 CoinJoins. The input/output ratio is 1.5x. So we are doing an upper bound with 2x.

Reply:
Now I am doubling the number of Alices that is not equal to the number of inputs. Alices seems to be the logical choice for me.